apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerRetryHelpersTests.cs

```diff
@@ -0,0 +1,249 @@
+﻿// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+using Google.Api.Gax.Grpc;
+using Google.Api.Gax.Testing;
+using Google.Protobuf;
+using Google.Protobuf.WellKnownTypes;
+using Google.Rpc;
+using Grpc.Core;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Xunit;
+using Status = Grpc.Core.Status;
+
+namespace Google.Cloud.Spanner.Data.Tests
+{
+    public class SpannerRetryHelpersTests
```

Comment:
Note: I haven't looked through the tests carefully, partly because I suspect they'll change based on other comments.